server/src/main/java/org/opensearch/transport/nativeprotocol/NativeOutboundMessage.java

```diff
@@ -48,16 +55,17 @@
  *
  * @opensearch.internal
  */
-abstract class OutboundMessage extends NetworkMessage {
+public abstract class NativeOutboundMessage extends NetworkMessage implements ProtocolOutboundMessage {
```

Comment:
```suggestion
abstract class NativeOutboundMessage extends NetworkMessage implements ProtocolOutboundMessage {
```

Reply:
It is needed for tests like InboundPipelineTests etc.

Reply:
Not really since InboundPipeline is common for all protocols, right now it uses native protocol things since it is default and would need references from that.

Reply:
>  right now it uses native protocol things since it is default and would need references from that.

I think it perfectly justifies the move, for protobuf we will add another test to `protobuf` package very likely

Reply:
That makes sense! Actually there are quite a few other classes that need to be moved to nativeprotocol package like InboundAggregator, InboundDecoder etc. They kind of go hand in hand with the tests. Do you mind if I open a new PR after this PR for them? It might involve more files.

Reply:
> Do you mind if I open a new PR after this PR for them? It might involve more files.

Sure, it is up to you, I think the moving is simple and having clean change would definitely be better I think

Reply:
Sure! Let me fix the 2 tests that are failing and I can add this change as well.